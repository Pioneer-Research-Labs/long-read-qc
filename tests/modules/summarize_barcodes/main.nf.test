nextflow_process {

    name "Test Process summarize_barcodes"
    script "modules/summarize_barcodes/main.nf"
    process "summarize_barcodes"

    test("Should generate barcode summaries and plots") {

        when {
            params {
                outdir = outputDir
            }
            process {
                """
                input[0] = file('${projectDir}/test_data/barcode_map.tsv')
                """
            }
        }

        then {
            assert process.success
            assert process.out.size() == 2
            assert process.out[0][0].size() == 4
            assert process.out[1][0].size() == 3
            assert snapshot("$outputDir/barcode_copy_number.csv",
            "$outputDir/barcode_lengths.csv",
            "$outputDir/barcode_proportions.csv",
            "$outputDir/concatenated_barcodes.csv",
            "$outputDir/barcode_copy_number.png",
            "$outputDir/barcode_length_distribution.png",
            "$outputDir/barcode_proportions.png",
            ).match()
        }

    }

}
