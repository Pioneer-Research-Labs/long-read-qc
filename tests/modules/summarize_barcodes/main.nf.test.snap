{
    "Should generate barcode summaries and plots": {
        "content": [
            "barcode_copy_number.csv:md5,3e653353445fddd7ee9082cccfc73de0",
            "barcode_lengths.csv:md5,994b6ee6428993ed190247f4f02cc346",
            "barcode_proportions.csv:md5,2cf6316949e0b7137484a047f6cb3d5d",
            "concatenated_barcodes.csv:md5,994b6ee6428993ed190247f4f02cc346",
            "barcode_copy_number.png:md5,08eb2433f4239b2d61f365aa962f2a11",
            "barcode_length_distribution.png:md5,f8c44064170b7112fd9ea6709767ebf7",
            "barcode_proportions.png:md5,e0fdef2ed85d10a2e125d59b99f0e118"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-15T13:35:18.92151"
    }
}